### ISSUE TITLE
Support for PHP 5.4.x

### ISSUE THREAD

**Issue by JasonBoyles** [rackspace-orchestration-templates/php-app-single#16]
From Jamie Hannaford:
> If offering 14.04+ is not an option, could we at least offer PHP 5.4+?

This is a stopgap option we could implement if 14.04 supported become complicated, but we need a reputable source for a package.

**Comment by JasonBoyles** [comment]
There is a `php5-oldstable` PPA which contains php 5.4, and it's maintained by the Debian folks who maintain current the current php5 packages:

https://launchpad.net/~ondrej/+archive/ubuntu/php5

> If you need to stay with PHP 5.4 you can use the oldstable PHP repository:
>
>    `ppa:ondrej/php5-oldstable`

Adding this PPA up front, before the php5 package is installed, may get us php 5.4

**Comment by JasonBoyles** [comment]
We can get php 5.4 on this server by building with the following cloud-init:

```yaml
#cloud-config
apt_sources:
 - source: "ppa:ondrej/php5-oldstable"
 package_update: true
```

**Comment by jamiehannaford** [comment]
@JasonBoyles yeah, that PPA is probably the best chance of getting a newer version of PHP on 12.04. Since we're using that, though, it probably makes sense to use PHP 5.5 (i.e. `ppa:ondrej/php5`). PHP 5.4 has entered an EOL period and only receives security updates. 5.5 is good until early 2016.

**Comment by jamiehannaford** [comment]
Hi @JasonBoyles - has there been any progress on this?

**Comment by JasonBoyles** [comment]
@jamiehannaford We have issues that are higher priority that are consuming our time right now. We will update this issue when we tackle the problem further.
